lib/api/objectDelete.js

```diff
@@ -73,6 +69,12 @@ function objectDelete(authInfo, request, log, cb) {
                     // versioning has been configured
                     return next(null, bucketMD, objMD);
                 }
+                // based on AWS's behavior, object delete can be performed by
+                // either bucket owner or the object owner
+                if (objMD['owner-id'] !== canonicalID
```

Comment:
The difference is subtle, but if you move this check above the `if (!objMD)` and add the `objMD && ` condition at the beginning (similar to what is done for `objectPut`), it allows to return `AccessDenied` whether the object exists or not, without leaking info about the object existence for unauthorized users. I don't know what AWS does in this respect but it could be good to check.

Reply:
Yes, corrected. Checked AWS behavior, you get access denied on non existing object.